### PR TITLE
Deprecate ThemeData.selectedRowColor

### PR DESCRIPTION
**Pull request by Piinks**
Related https://github.com/flutter/flutter/issues/91772

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -1796,7 +1808,11 @@ class ThemeData with Diagnosticable {
       'https://flutter.dev/docs/release/breaking-changes/toggleable-active-color#migration-guide. '
       'This feature was deprecated after v2.13.0-0.4.pre.',
     )
-    Color? toggleableActiveColor,
+    Color? toggleableActiveColor,@Deprecated(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: newline needed here?

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/theme_data.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/material/theme_data.dart]: I think one was added while I was reviewing. Please ignore this.

Comment (Piinks) [reply packages/flutter/lib/src/material/theme_data.dart]: 😉

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -1883,15 +1899,16 @@ class ThemeData with Diagnosticable {
...
       tooltipTheme: tooltipTheme ?? this.tooltipTheme,
       // DEPRECATED (newest deprecations at the bottom)
-      accentColor: accentColor ?? this.accentColor,
-      accentColorBrightness: accentColorBrightness ?? this.accentColorBrightness,
-      accentTextTheme: accentTextTheme ?? this.accentTextTheme,
-      accentIconTheme: accentIconTheme ?? this.accentIconTheme,
-      buttonColor: buttonColor ?? this.buttonColor,
-      fixTextFieldOutlineLabel: fixTextFieldOutlineLabel ?? this.fixTextFieldOutlineLabel,
-      primaryColorBrightness: primaryColorBrightness ?? this.primaryColorBrightness,
+      accentColor: accentColor ?? _accentColor,
```
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:1902]: Added this tweak. `this.<deprecatedProperty>` here does not actually allow folks to stop using it. I just found out in a test case. 😜

**Comment by Piinks** [comment]
Will see about landing this next week, I want to run it through internal tests first.

**Comment by Piinks** [comment]
Internal testing passed.

**Pull request merged**